op-chain-ops/cmd/registry-data/main.go

```diff
@@ -91,17 +91,19 @@ type GenesisAccount struct {
 }
 
 type Genesis struct {
```

Comment:
is there any reason as to why we need to copy/paste this out of geth? because it doesn't have json encoding? it will continue to get out of date, ie prague is adding new fields to block header

Reply:
Agreed that this will continue to get out of date. 

We can't use the types directly from geth because what is being crafted here is actually a kind of "compressed" version of the genesis, (e.g. replacing bytecode with a hash) https://github.com/ethereum-optimism/superchain-registry/blob/main/CONTRIBUTING.md#3-understand-output

The solution I would like to advocate for, to keep everything up to date, is to have the tool
* read in the source `genesis.json` file (as it does now)
* compress and store it in the superchain registry (as it does now)
* immediately uncompress it, using code that should be _colocated_ with this tool.
    - https://github.com/ethereum-optimism/superchain-registry/blob/1a6df3f940537bb497d88055861a77295ee9e827/superchain/superchain.go#L557
    - https://github.com/ethereum-optimism/op-geth/blob/9617f2a7dd0ab80c468d6e474e3cc26be9572c97/core/superchain.go#L13-L14
* check for deep equality with the input `genesis.json` file
* panic if there is a diff, at which point we can address the mismatch.

Reply:
This approach seems good to me